pkg/controller/pod_volume_backup_controller.go

```diff
@@ -320,7 +320,12 @@ func getParentSnapshot(log logrus.FieldLogger, pvcUID string, podVolumeBackupLis
 			continue
 		}
 
-		if mostRecentBackup == nil || backup.Status.StartTimestamp.After(mostRecentBackup.Status.StartTimestamp.Time) {
+		if (mostRecentBackup == nil || backup.Status.StartTimestamp.After(mostRecentBackup.Status.StartTimestamp.Time)) &&
+			// Check the backup storage location is the same as spec in order to support backup to multiple backup-locations.
+			// Otherwise, there exist a case that backup volume snapshot to the second location would failed, since the founded
+			// parent ID is valid for the first backup location, not the second backup location.
+			// Also, the second backup should not use first backup parent ID since eachthe parent ID is for the first backup location only.
+			(backupStorageLocation == backup.Spec.BackupStorageLocation) {
```

Comment:
I'd prefer to move this new check into its own guard condition just above L323, so basically:

```go
if backup.Spec.BackupStorageLocation != backupStorageLocation {
    continue
}
```

This should make it a little easier to read.

It would also be nice to add a unit test here, but I'm not going to block merge on it given that we don't already have one in place to easily add new case to.

Reply:
Update for easier to read.